### PR TITLE
Pivotal ID # 185967206: Backend text search capabilities

### PR DESCRIPTION
**Pull request by Juan-EBI**
https://www.pivotaltracker.com/n/projects/1189310

### CODE REVIEW

**Review by jhoanmanuelms — COMMENTED** [review]
LGTM. Please consider my comments.

Also, there's one concern I have which is related to the remove changelogs. Since this PR will leave the DB in a clean state and it's been a while since we deployed to PROD, I think we should first deploy PROD, to make sure the current changelogs are executed and then merge this change.

WDYT?

**Comment by Juan-EBI** [comment]
> LGTM. Please consider my comments.
> 
> Also, there's one concern I have which is related to the remove changelogs. Since this PR will leave the DB in a clean state and it's been a while since we deployed to PROD, I think we should first deploy PROD, to make sure the current changelogs are executed and then merge this change.
> 
> WDYT?

We can be 100% sure of what are the changes has been executed already. Is as simple as check eubioimag
.submitter_mongockChangeLog collection. I already check and last one is the number 009 which is the last one before this changes.

File: submission/persistence-mongo/src/main/kotlin/ac/uk/ebi/biostd/persistence/doc/migrations/DatabaseChangeLog.kt [hunk 1]
```diff
@@ -26,216 +30,161 @@ import ac.uk.ebi.biostd.persistence.doc.db.converters.shared.FileListDocFileFiel
...
+                .on("$prefix$SUB_COLLECTIONS.$COLLECTION_ACC_NO", ASC)
+                .on("$prefix$SUB_VERSION", ASC)
+                .on("$prefix$STORAGE_MODE", ASC)
+        )
+        ensureIndex(
+            TextIndex()
+                .onField("$prefix$SUB_TITLE")
+                .onField("$prefix$SUB_SECTION.$SEC_ATTRIBUTES.$ATTRIBUTE_DOC_NAME")
+                .onField("$prefix$SUB_SECTION.$SEC_ATTRIBUTES.$ATTRIBUTE_DOC_VALUE")
+                .named("title_text_section.attributes.name_text_section.attributes.value_text")
```
[the remark this answers is not in the document]
Comment (Juan-EBI) [reply submission/persistence-mongo/src/main/kotlin/ac/uk/ebi/biostd/persistence/doc/migrations/DatabaseChangeLog.kt:102]: this is the index already created so try to created it with another name will fail

File: submission/persistence-mongo/src/main/kotlin/ac/uk/ebi/biostd/persistence/doc/migrations/DatabaseChangeLog.kt [hunk 2]
```diff
@@ -26,216 +30,161 @@ import ac.uk.ebi.biostd.persistence.doc.db.converters.shared.FileListDocFileFiel
...
-
-        template.updateMulti(Query(), Update().unset(SUB_STATS), DocSubmission::class.java)
+/**
+ * submission_requests collection Indexes
+ * 1. AccNo
+ * 2. AccNo - Version
+ * 3. All submission index over internal submission.
+ */
+fun MongoOperations.ensureSubmissionRequestIndexes() {
+    ensureSubmissionIndexes<DocSubmissionRequest>("$SUB.")
```
[the remark this answers is not in the document]
Comment (Juan-EBI) [reply submission/persistence-mongo/src/main/kotlin/ac/uk/ebi/biostd/persistence/doc/migrations/DatabaseChangeLog.kt:115]: agree we can not garantee both will always have exacttly the same indices, but right now the index are exactly the same which make sense as we want to search submission and submission request by the same fields both appear together in the submission tool, I do not see in which cases we will not want that.

Said if indexes start being diferent  we can separate functions, that is abolutly fine, but given current conditions I think having them together make total sence.  I prefer the best design for current requirements. Is not that I am worry about duplicated code  I can easily separte them but I think that will faciliate that we put an index in one part and not in the other that given current conditions (we search them together) will be a bug. Basically I want to make harder to introduce a bug

**Review by jhoanmanuelms — APPROVED** [review]
LGTM

**Pull request merged**